engine/pkg/client/executor_group.go

```diff
@@ -202,8 +257,27 @@ func (g *DefaultExecutorGroup) RemoveExecutor(executorID model.ExecutorID) error
 	client.Close()
 
 	delete(g.clients, executorID)
```

Comment:
Seems that `g.client` and `g.tomstoneList` is updated under a lock `g.mu`, checking the `online list` and `offline list` has no difference. 

Can you explain more about the `executor has gone offline`?

Reply:
when an executor's heartbeat is timeout, I think we no longer need to send something to executor.
So `executor is offline` and `executor has gone offline` seems have no big difference.

Reply:
- For situation 1, we need to give up immediately and try another executor.
- For situation 2, we need to wait for a while and retry.

Reply:
Get the point.
`executor online list` in servermaster and executor may be different for some network delay.
So when a jobmaster get a worker scheduler plan with an executor id in it, it may find the `oriented executor` is not in the its `online executor list`. In that case, we need a tomstone list to indicate which executor is dead certainly.

Reply:
We can add some comments to avoid the unexpected modification on this mechanism in the future.